src/en/kami18/src/eu/kanade/tachiyomi/extension/en/kami18/Kami18.kt

```diff
@@ -0,0 +1,178 @@
+package eu.kanade.tachiyomi.extension.en.kami18
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.lang.Exception
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Kami18() : HttpSource() {
+
+    override val name = "18Kami"
+
+    override val lang = "en"
+
+    override val baseUrl = "https://18kami.com"
+
+    private val baseImageUrl = "$baseUrl/media/photos"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient
+
+    override fun headersBuilder() = Headers.Builder().apply {
+        add("Referer", "$baseUrl/")
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/albums?o=mv&page=$page", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()

```

Comment:
doesn't really matter, since `ParsedHttpSource` can sometimes require overriding lot more useless methods. e.g overriding `searchMangaRequest` to point to `popularMangaRequest` would still require overriding three other mothods.